week-8/Homework/mandatory/3-dog-photo-gallery/script.js

```diff
@@ -0,0 +1,22 @@
+let nextImg = document.getElementById("next");
+
+nextImg.addEventListener("click", function () {
+  fetch("https://dog.ceo/api/breeds/image/random")
+    .then((response) => {
+      if (response.status >= 200 && response.status <= 299) {
+        return response.json();
```

Comment:
very good work here, I'd say there is no 299 code
so something between 200 and 304 is ideal in the most situations
also for this error catching in the first promise. the response body was not parsed yet to json. Most API's add error bodies to its response. 
What I mean, in your code, the statusText is probably empty all the time, so it only returns 404
but if you waited until the response get parsed you may get the message the API will send with the error.. try soemthing like 
`  .then((response) => {
     
      return response.json();
    })
    .then((response) => {
      if (response.status == "success") {
        let dogImgList = document.getElementById("dogList");
        dogImgList.innerHTML += `<li><img src="${response.message}"></li>`;
        console.log(response.message);
      } else {
        throw new Error(
          `Encountered something unexpected: ${response.status} ${response.message}`
        );
      }
    })`